pkg/services/pluginsintegration/angularpatternsstore/store.go

```diff
@@ -2,74 +2,42 @@ package angularpatternsstore
 
 import (
 	"context"
-	"encoding/json"
-	"fmt"
-	"time"
 
 	"github.com/grafana/grafana/pkg/infra/kvstore"
+	"github.com/grafana/grafana/pkg/services/pluginsintegration/cachekvstore"
 )
 
 type Service interface {
+	cachekvstore.LastUpdateGetter
 	Get(ctx context.Context) (string, bool, error)
-	Set(ctx context.Context, patterns any) error
-	GetLastUpdated(ctx context.Context) (time.Time, error)
+	Set(ctx context.Context, value any) error
 }
 
 const (
 	kvNamespace = "plugin.angularpatterns"
-
-	keyPatterns    = "angular_patterns"
-	keyLastUpdated = "last_updated"
+	keyPatterns = "angular_patterns"
 )
 
 // KVStoreService allows to cache GCOM angular patterns into the database, as a cache.
 type KVStoreService struct {
-	kv *kvstore.NamespacedKVStore
+	*cachekvstore.NamespacedStore
 }
 
+var _ Service = (*KVStoreService)(nil)
+
 func ProvideService(kv kvstore.KVStore) Service {
 	return &KVStoreService{
-		kv: kvstore.WithNamespace(kv, 0, kvNamespace),
+		NamespacedStore: cachekvstore.NewNamespacedStore(kv, kvNamespace),
 	}
 }
 
-// Get returns the raw cached angular detection patterns. The returned value is a JSON-encoded string.
-// If no value is present, the second argument is false and the returned error is nil.
+// Get returns the stored angular patterns from the underlying cachekvstore.
 func (s *KVStoreService) Get(ctx context.Context) (string, bool, error) {
-	return s.kv.Get(ctx, keyPatterns)
-}
-
-// Set sets the cached angular detection patterns and the latest update time to time.Now().
-// patterns must implement json.Marshaler.
-func (s *KVStoreService) Set(ctx context.Context, patterns any) error {
-	b, err := json.Marshal(patterns)
-	if err != nil {
-		return fmt.Errorf("json marshal: %w", err)
-	}
-	if err := s.kv.Set(ctx, keyPatterns, string(b)); err != nil {
-		return fmt.Errorf("kv set: %w", err)
-	}
-	if err := s.kv.Set(ctx, keyLastUpdated, time.Now().Format(time.RFC3339)); err != nil {
-		return fmt.Errorf("kv last updated set: %w", err)
-	}
-	return nil
+	return s.NamespacedStore.Get(ctx, keyPatterns)
 }
 
-// GetLastUpdated returns the time when Set was last called. If the value cannot be unmarshalled correctly,
-// it returns a zero-value time.Time.
-func (s *KVStoreService) GetLastUpdated(ctx context.Context) (time.Time, error) {
-	v, ok, err := s.kv.Get(ctx, keyLastUpdated)
-	if err != nil {
-		return time.Time{}, fmt.Errorf("kv get: %w", err)
-	}
-	if !ok {
-		return time.Time{}, nil
-	}
-	t, err := time.Parse(time.RFC3339, v)
-	if err != nil {
-		// Ignore decode errors, so we can change the format in future versions
-		// and keep backwards/forwards compatibility
-		return time.Time{}, nil
-	}
-	return t, nil
+// Set stores the given angular patterns in the underlying cachekvstore.
+// TODO: change signature so `value` accepts only the correct type?
```

Comment:
do you want to address this TODO? (either that or remove the comment)

Reply:
I tried, but I forgot why I used `any` in the first place:

```go
package github.com/grafana/grafana/pkg/cmd/grafana
	imports github.com/grafana/grafana/pkg/cmd/grafana-cli/commands
	imports github.com/grafana/grafana/pkg/cmd/grafana-cli/commands/secretsmigrations
	imports github.com/grafana/grafana/pkg/server
	imports github.com/grafana/grafana/pkg/registry/backgroundsvcs
	imports github.com/grafana/grafana/pkg/services/pluginsintegration/angulardetectorsprovider
	imports github.com/grafana/grafana/pkg/services/pluginsintegration/angularpatternsstore
	imports github.com/grafana/grafana/pkg/services/pluginsintegration/angulardetectorsprovider: import cycle not allowed
```

So I will leave it for a follow-up PR eventually